src/components/application_manager/src/request_info.cc

```diff
@@ -116,6 +116,15 @@ FakeRequestInfo::FakeRequestInfo(uint32_t app_id, uint32_t correaltion_id) {
   correlation_id_ = correaltion_id;
 }
 
+RequestInfoSet::~RequestInfoSet() {
+  sync_primitives::AutoLock lock(pending_requests_lock_);
+  auto it = time_sorted_pending_requests_.begin();
+  while (!time_sorted_pending_requests_.empty()) {
+    RemoveRequest(*it);
```

Comment:
seems this would cause a double `Acquire` for `pending_requests_lock_` since it isn't recursive

Reply:
```suggestion
    Erase(*it);
```
Seems this would fix the issue